diff --git a/node_modules/react-native-reanimated/apple/reanimated/apple/keyboardObserver/REAKeyboardEventObserver.mm b/node_modules/react-native-reanimated/apple/reanimated/apple/keyboardObserver/REAKeyboardEventObserver.mm
index 5473fdf..990ab3d 100644
--- a/node_modules/react-native-reanimated/apple/reanimated/apple/keyboardObserver/REAKeyboardEventObserver.mm
+++ b/node_modules/react-native-reanimated/apple/reanimated/apple/keyboardObserver/REAKeyboardEventObserver.mm
@@ -330,7 +330,6 @@ - (void)keyboardWillHide:(NSNotification *)notification
 - (void)updateKeyboardHeightDuringInteractiveDismiss:(CGPoint)oldKeyboardFrame
                                     newKeyboardFrame:(CGPoint)newKeyboardFrame
 {
-  _isInteractiveDismissalCanceled = false;
   auto keyboardView = [self getKeyboardView];
   bool hasKeyboardAnimation = [self hasAnyAnimation:keyboardView];
   if (hasKeyboardAnimation) {
@@ -342,6 +341,7 @@ - (void)updateKeyboardHeightDuringInteractiveDismiss:(CGPoint)oldKeyboardFrame
     return;
   }
   float visibleKeyboardHeight = windowHeight - (newKeyboardFrame.y - keyboardHeight / 2);
+  _isInteractiveDismissalCanceled = false;
   if (oldKeyboardFrame.y > newKeyboardFrame.y) {
     _state = OPENING;
     _isInteractiveDismissalCanceled = true;
